### PR TITLE
Re-writes kvm docs for greater technical accuracy

### PR DESCRIPTION
**Pull request by iphutch**
Signed-off-by: ihutchin <ianeta.hutchinson@intel.com>

### CODE REVIEW

**Comment by rcaballeromx** [comment]
@miguelinux: Can you take a look and confirm the content is accurate and give us your opinion? Thanks.

File: source/clear-linux/virtual-environments/kvm.rst [hunk 1]
```diff
@@ -1,57 +1,156 @@
...
-(Intel® VT) and `Intel® Virtualization Technology for Directed I/O
-<https://software.intel.com/en-us/articles/intel-virtualization-technology-for-directed-io-vt-d-enhancing-intel-platforms-for-efficient-virtualization-of-io-devices>`_
-(Intel® VT-d) in your BIOS/UEFI firmware configuration.
+1. Enable `Intel® Virtualization Technology`_ (Intel® VT) and
+   `Intel®Virtualization Technology for Directed I/O`_ (Intel® VT-d) in the
+   host machine’s BIOS.
 
-Starter script
-==============
+2. Log in and get root privilege on the host machine::
```
Comment (rcaballeromx) [inline source/clear-linux/virtual-environments/kvm.rst]: Please do not use the :: for commands. Use the code-block directive with the appropriate language, in this case console. Additionally, do not use explicit numbering for ordered lists. use #.
These markup comments apply to the entire document.

File: source/clear-linux/virtual-environments/kvm.rst [hunk 2]
```diff
@@ -1,57 +1,156 @@
...
+
+9. Start the |CL| KVM virtual machine::
+
+   # ./start_qemu.sh clear-<version number>-kvm.img
+
+#. Log in and set the root password.
+
+#. To SSH into the |CL| VM, follow these steps.
+
+    a. Enable SSH access
```
Comment (rcaballeromx) [inline source/clear-linux/virtual-environments/kvm.rst]: Use the #. for the second level ordered list too. You do have to be careful with the indentation to ensure that the list does not break. But using #. allows us to add or remove steps without having to modify all the numbers manually.

File: source/clear-linux/virtual-environments/kvm.rst [hunk 3]
```diff
@@ -1,57 +1,156 @@
...
 
-Using KVM
-#########
+Running Clear Linux as a KVM guest OS
+#####################################
 
-The easiest way to get started in a virtualized environment is to download
-a recent KVM image from the `images`_ directory. Here you'll find a kvm
-image file, the UEFI firmware helper, and the KVM start helper script.
+To run |CL| in a virtualized environment using abbr:`KVM (Kernel-based
```
Comment (rcaballeromx) [inline source/clear-linux/virtual-environments/kvm.rst]: Change to |CLOSIA| to have the full trademarked name at least once on the document.

File: source/clear-linux/virtual-environments/kvm.rst [hunk 4]
```diff
@@ -1,57 +1,156 @@
...
+
+   # chmod +x start_qemu.sh
+
+9. Start the |CL| KVM virtual machine::
+
+   # ./start_qemu.sh clear-<version number>-kvm.img
+
+#. Log in and set the root password.
+
+#. To SSH into the |CL| VM, follow these steps.
```
Comment (rcaballeromx) [inline source/clear-linux/virtual-environments/kvm.rst]: follow these steps:

File: source/clear-linux/virtual-environments/kvm.rst [hunk 5]
```diff
@@ -1,57 +1,156 @@
...
+
+       # ssh -p 10022 root@localhost
+
+#. To add the GNOME Display Manager (GDM) to the |CL| VM, follow these steps:
+
+   a. Shutdown the active |CL| VM.
+
+   b. Install VNCViewer on the host machine.  Below are some example distros.
+
+      *  Clear Linux::
```
Comment (rcaballeromx) [inline source/clear-linux/virtual-environments/kvm.rst]: Preface each distro with the preposition On. i.e. On Clear Linux, On Fedora 25 Workstation, etc.

File: source/clear-linux/virtual-environments/kvm.rst [hunk 6]
```diff
@@ -1,57 +1,156 @@
...
+
+      *  Mint 18.1 “Serena” Desktop::
+
+         # apt-get vncviewer
+
+      *  Fedora 25 Workstation::
+
+         # dnf install tigervnc
+
+   c. Modify the start_qemu.sh script to increase memory (-m), add graphics
```
Comment (rcaballeromx) [inline source/clear-linux/virtual-environments/kvm.rst]: Use the :file: role to markup the filename. :file:\`start_qemu.sh\`

File: source/clear-linux/virtual-environments/kvm.rst [hunk 7]
```diff
@@ -1,57 +1,156 @@
...
+             -vga qxl \
+             -vnc :0 -nographic \
+             -usbdevice tablet \
+             -drive file="$IMAGE",if=virtio,aio=threads,format=raw \
+             -netdev user,id=mynet0,hostfwd=tcp::${VMN}0022-
+             :22,hostfwd=tcp::${VMN}2375-:2375 \
+             -device virtio-net-pci,netdev=mynet0 \
+             -debugcon file:debug.log -global isa-debugcon.iobase=0x402 $@
+
+   d. Due to changes in `start_qemu.sh` script, the UEFI NvVars information for
```
Comment (rcaballeromx) [inline source/clear-linux/virtual-environments/kvm.rst]: Ditto

File: source/clear-linux/virtual-environments/kvm.rst [hunk 8]
```diff
@@ -1,57 +1,156 @@
...
-   $ bash start_qemu.sh clr_image &
+     # dnf install qemu-kvm
 
-*  If you'd like to run the script with execute permission::
+4. Download the latest pre-built Clear Linux _`KVM image` file from
+   the _`image` directory.
 
-   $ chmod +x start_qemu.sh
-   $ ./start_qemu.sh clr_image
+5. Uncompress the downloaded image
```
Comment (bktan8) [inline source/clear-linux/virtual-environments/kvm.rst]: Missing a punctuation at the end.

File: source/clear-linux/virtual-environments/kvm.rst [hunk 9]
```diff
@@ -1,57 +1,156 @@
...
 
-*  And to run it as a background process::
+   .. code-block:: console
 
-   $ ./start_qemu.sh clr_image &
+      unxz clear-<version number>-kvm.img.xz
```
Comment (bktan8) [inline source/clear-linux/virtual-environments/kvm.rst]: Missing the '#' in front of the command.

File: source/clear-linux/virtual-environments/kvm.rst (hunk 7 above)
Comment (bktan8) [inline source/clear-linux/virtual-environments/kvm.rst]: For some reason, this line is in bold and italicized when rendered.

**Comment by iphutch** [comment]
@rcaballeromx @bktan8 changes applied. Can't seem to make `images`_ link. Will make WIP until that's fixed

**Comment by iphutch** [comment]
@bktan8 all changes have been applied and links fixed.
@rcaballeromx I used in line links for `images` as I faced many errors regarding duplicate target names. Please edit if inappropriate.

**Comment by iphutch** [comment]
@bktan8 please review and approve if appropriate.

**Comment by rcaballeromx** [comment]
@miguelinux Can you take a look at this PR too.

File: source/clear-linux/virtual-environments/kvm.rst [hunk 10]
```diff
@@ -3,60 +3,80 @@
...
-1. Enable `Intel® Virtualization Technology`_ (Intel® VT) and
+Install KVM
+===========
+
+#. Enable `Intel® Virtualization Technology`_ (Intel® VT) and
    `Intel®Virtualization Technology for Directed I/O`_ (Intel® VT-d) in the
    host machine’s BIOS.
 
-2. Log in and get root privilege on the host machine::
+#. Log in and get root privilege on the host machine:
```
Comment (miguelinux) [inline source/clear-linux/virtual-environments/kvm.rst]: could we said

#. Log in **open a text terminal** and get root privilege on the host machine:

?

File: source/clear-linux/virtual-environments/kvm.rst [hunk 11]
```diff
@@ -1,57 +1,156 @@
...
-.. _images: http://download.clearlinux.org/image/
+8. Make the script executable::
+
+   # chmod +x start_qemu.sh
+
+9. Start the |CL| KVM virtual machine::
+
+   # ./start_qemu.sh clear-<version number>-kvm.img
+
+#. Log in and set the root password.
```
Comment (miguelinux) [inline source/clear-linux/virtual-environments/kvm.rst]: #. Log in as **root** user and set a new password

File: source/clear-linux/virtual-environments/kvm.rst [hunk 12]
```diff
@@ -3,60 +3,80 @@
...
-   # chmod +x start_qemu.sh
+      # chmod +x start_qemu.sh
 
-9. Start the |CL| KVM virtual machine::
+#. Start the |CL| KVM virtual machine:
 
-   # ./start_qemu.sh clear-<version number>-kvm.img
+     .. code-block:: console
+
+        # ./start_qemu.sh clear-<version number>-kvm.img
```
Comment (miguelinux) [inline source/clear-linux/virtual-environments/kvm.rst]: to me, I willl add:

With this script QEMU-KVM will launch a VM and give you the VM console input in the same terminal.

File: source/clear-linux/virtual-environments/kvm.rst [hunk 13]
```diff
@@ -3,60 +3,80 @@
...
 
 #. Log in and set the root password.
 
-#. To SSH into the |CL| VM, follow these steps.
+#. To SSH into the |CL| VM, follow these steps:
```
Comment (miguelinux) [inline source/clear-linux/virtual-environments/kvm.rst]: I put this info in a new sub-section

SSH into the new |CL| VM
----------------------------

You can do SSH login to your new |CL| VM. To do so, you need to launch the VM and follow these steps:

File: source/clear-linux/virtual-environments/kvm.rst [hunk 14]
```diff
@@ -144,32 +149,26 @@ To add the GNOME Display Manager (GDM) to the |CL| VM, follow these steps:
...
           -debugcon file:debug.log -global isa-debugcon.iobase=0x402 $@
 
-Reset UEFI NvVars information
-=============================
+#. Due to changes in :file:`start_qemu.sh` script, the UEFI :file:`NvVars`
+   information for the previously-booted |CL| VM will need to be reset.
 
-Due to changes in :file:`start_qemu.sh` script, the UEFI :file:`NvVars`
-information for the previously-booted |CL| VM will need to be reset.
+   #. Relaunch the |CL| VM.  The EFI shell will appear:
```
Comment (miguelinux) [inline source/clear-linux/virtual-environments/kvm.rst]: EFI or UEFI shell?

**Comment by iphutch** [comment]
@miguelinux if you could please re-review. Bun has seen to all your comments :)

**Comment by iphutch** [comment]
I have submitted a new PR that has no merge conflicts:
https://github.com/01org/clear-linux-documentation/pull/36 
@miguelinux @rcaballeromx @bktan8

**Pull request closed without merging**